src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CfEnvAwareResource.java

```diff
@@ -145,6 +145,10 @@ static boolean hasCfEnv(CfEnvAwareResource app
 					Class.forName(CF_ENV, false, classLoader);
 					return true;
 				}
+				catch (UnsupportedClassVersionError err) {
+					// class found but can't load it i.e. because it's newer class version
```

Comment:
It may be helpful to add a `logger.trace` here for later possible debugging if ever needed. Wdyt?

Reply:
Yeah